src/v/redpanda/admin_server.h

```diff
@@ -113,11 +136,16 @@ class admin_server {
               // from authenticate().
               log_request(*req, auth_state);
 
-              if constexpr (peek_auth) {
-                  return handler(std::move(req), auth_state);
-              } else {
-                  return handler(std::move(req));
-              }
+              return intercept_exceptions(
+                req->get_url(),
+                auth_state,
```

Comment:
if you change intercept_exceptions to co_await the handler (see other comment), pay close attention to whether it's still safe to pass auth_state (stack var) by reference into this function (reference may be invalid after a co_await).